src/System.Private.CoreLib/src/System/AppContext.Unix.cs

```diff
@@ -17,7 +19,34 @@ public static string BaseDirectory
         {
             get
             {
-                throw new NotImplementedException();
+                int currentSize = Path.MaxPath;
```

Comment:
Path.MaxPath on Unix tends to be huge. It is better to start with something reasonable like 256, and grow it from there. Take a look at existing similar methods like GetCwd.